src/Symfony/Component/Form/Extension/DataCollector/FormDataCollector.php

```diff
@@ -221,6 +238,85 @@ public function getData()
         return $this->data;
     }
 
+    public function serialize()
+    {
+        $cloneVar = array($this, 'cloneVar');
+
+        foreach ($this->data['forms_by_hash'] as &$form) {
+            if ($form['type_class'] instanceof Data) {
+                continue;
+            }
+            $form['view_vars'] = array_map($cloneVar, $form['view_vars']);
+            $form['type_class'] = $cloneVar(new ClassStub($form['type_class']));
+            $form['synchronized'] = $cloneVar($form['synchronized']);
+            $form['passed_options'] = array_map($cloneVar, $form['passed_options']);
+            $form['resolved_options'] = array_map($cloneVar, $form['resolved_options']);
+            $form['default_data'] = array_map($cloneVar, $form['default_data']);
+            $form['submitted_data'] = array_map($cloneVar, $form['submitted_data']);
+
+            if (!empty($form['errors'])) {
+                foreach ($form['errors'] as $i => $error) {
+                    if (!empty($error['trace'])) {
+                        $form['errors'][$i]['trace'] = array_map($cloneVar, $error['trace']);
+                    }
+                }
+            }
+        }
+
+        return serialize($this->data);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function cloneVar($var)
+    {
+        if (null === $this->cloner) {
+            if (class_exists(ClassStub::class)) {
+                $this->cloner = new VarCloner();
+                $this->cloner->setMaxItems(25);
+                $this->cloner->addCasters(array(
+                    Stub::class => function (Stub $v, array $a, Stub $s, $isNested) {
+                        return $isNested ? $a : StubCaster::castStub($v, $a, $s, true);
+                    },
+                    \Exception::class => function (\Exception $e, array $a, Stub $s) {
+                        if (isset($a[$k = "\0Exception\0previous"])) {
+                            unset($a[$k]);
+                            ++$s->cut;
+                        }
+
+                        return $a;
+                    },
+                    FormInterface::class => function (FormInterface $f, array $a) {
+                        return array(
+                            Caster::PREFIX_VIRTUAL.'name' => $f->getName(),
+                            Caster::PREFIX_VIRTUAL.'type_class' => new ClassStub(get_class($f->getConfig()->getType()->getInnerType())),
+                        );
+                    },
+                    ConstraintViolationInterface::class => function (ConstraintViolationInterface $v, array $a) {
+                        return array(
+                            Caster::PREFIX_VIRTUAL.'root' => $v->getRoot(),
+                            Caster::PREFIX_VIRTUAL.'path' => $v->getPropertyPath(),
+                            Caster::PREFIX_VIRTUAL.'value' => $v->getInvalidValue(),
```

Comment:
Just checked [in the original code](https://github.com/symfony/symfony/pull/19614/files#diff-1b203840c96e2f4a2f0386788e2765d3L117). This just mirrors it. LGTM.